src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -100,7 +103,9 @@ export const UnderlineNavItem = forwardRef(
 
         setChildrenWidth({text, width: domRect.width})
         setNoIconChildrenWidth({text, width: domRect.width - iconWidthWithMargin})
-        preSelected && selectedLink === undefined && setSelectedLink(ref as RefObject<HTMLElement>)
+
+        if (selectedLink === undefined && ariaCurrent !== undefined && ariaCurrent !== 'false' && ariaCurrent !== false)
+          setSelectedLink(ref as RefObject<HTMLElement>)
```

Comment:
[Minor] I like to use curly braces for multi-line if statements
```suggestion
        if (selectedLink === undefined && ariaCurrent !== undefined && ariaCurrent !== 'false' && ariaCurrent !== false) {
          setSelectedLink(ref as RefObject<HTMLElement>)
        }
```